packages/babel-plugin-transform-react-constant-elements/test/fixtures/constant-elements/append-to-end-when-declared-in-scope-3/output.js

```diff
@@ -17,10 +17,13 @@ var _ref3 =
   const AppItem = () => {
     return _ref2;
   },
+        _ref4 =
+  /*#__PURE__*/
+  (() => <AppItem />)(),
         _ref =
   /*#__PURE__*/
-  <div>
+  (() => <div>
           {_ref3}
-          <AppItem />
-        </div>;
+          {_ref4}
```

Comment:
The `<AppItem />` here is unnecessarily hoisted to the same scope of `_ref` because the Path Hoister can not neglect the introduced IIAFE. Note that when the JSX element is _not_ hoisted to the top level, its children may has opportunity hoisted to top level. But we can't know which scope the element will be hoisted to until we have run `path.hoist()`, after that the AST is changed and it's too later to determine whether such hoisting is favourable.

Reply:
What if we added a new parameter to `path.hoist()`, to say "hoist this path but only if it can go at least up to this scope"?

Reply:
Yeah, we can do it but due to the semver requirements we can not ship it in patch/minor releases unless we are doing dirty version checks.

Do we have plans to change semver to minor-version compatibilities in Babel 8?

Reply:
If someone uses an old `@babel/traverse` version, their JSX elements will be hoisted singularly because the "minimum scope" parameter is ignored. It's not optimal, but it will still perfectly work.